src/components/ComicBook/MathQuestions.tsx

```diff
@@ -0,0 +1,142 @@
+import { Page, Question } from "../../types";
+import { useEffect, useState } from "react";
+import { SyntheticEvent } from "react";
+import comicService from "../../services/comicService";
+import axios from "axios";
+const MathQuestions = ({
+  stateKey,
+  setKey,
+  comicName,
+  comic,
+  page,
+}: {
+  stateKey: string | undefined;
+  setKey: React.Dispatch<React.SetStateAction<string | undefined>>;
+  comicName: string;
+  comic: Page[];
+  page: number;
+}) => {
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [answers, setAnswers] = useState<string[]>([""]);
+  const [buttonClasses, setButtonClasses] = useState("answer-button");
+
+  useEffect(() => {
+    let newQuestions: Question[] = [];
+    if (comic[page].questionList) {
+      const l = comic[page].questionList.length;
+      for (let q = 0; q < l; q++) {
+        const question = comic[page].questionList[q];
+        newQuestions = [...newQuestions, question];
+      }
+      setQuestions(newQuestions);
+      setAnswers(new Array(l).fill(""));
+    } else {
+      setQuestions(newQuestions);
+    }
+  }, [page, comic]);
+
+  const handleAnswer = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (
+      comic[page].questionList &&
```

Comment:
Maybe this `if (...)` can be cleand? To own function 
```
if(thereAreAnswers(comic[page]) {
...
```

Reply:
typescript says that questionList might be undefined, if I move these to a separate function. I did that, though. Now it just checks in two places if the same questionList exists: 
in that if(...), and in the function thereAreAnswers().